### PR TITLE
Contact/shared pointer update and cleanup

### PR DESCRIPTION
**Pull request by loumalouomega**
Concerning #1328

### CODE REVIEW

**Review by RiccardoRossi — DISMISSED** [review]

**Comment by loumalouomega** [comment]
Thanks

**Comment by loumalouomega** [comment]
I am having a compilation error in Release, I will keep you informed

**Comment by loumalouomega** [comment]
Now it works it was a problem of RAM, sorry for the noise

**Comment by loumalouomega** [comment]
@RiccardoRossi  I updated the solution of the two tests that crash during the night, I still don't know why they crashh, in my machine it works well both FullDebug and Release

**Comment by loumalouomega** [comment]
@RiccardoRossi PING

**Comment by loumalouomega** [comment]
@RiccardoRossi PING

File: applications/ContactStructuralMechanicsApplication/custom_strategies/custom_strategies/residualbased_newton_raphson_contact_strategy.h [hunk 1]
```diff
@@ -103,7 +103,7 @@ class ResidualBasedNewtonRaphsonContactStrategy :
     
     typedef ModelPart::ConditionsContainerType                                ConditionsArrayType;
     
-    typedef boost::shared_ptr<ProcessFactoryUtility>                            ProcessesListType;
+    typedef ProcessFactoryUtility::Pointer                                      ProcessesListType;
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/custom_strategies/custom_strategies/residualbased_newton_raphson_contact_strategy.h:106]: shouldn't this be `Kratos::shared_ptr<ProcessFactoryUtility>`?

**Review by philbucher — APPROVED** [review]
Changes look ok, I cannot check the tests though, but I assume you know what you are doing :)

File: applications/ContactStructuralMechanicsApplication/custom_strategies/custom_strategies/residualbased_newton_raphson_contact_strategy.h (hunk 1 above)
Comment (jcotela) [reply applications/ContactStructuralMechanicsApplication/custom_strategies/custom_strategies/residualbased_newton_raphson_contact_strategy.h:106]: @philbucher it is exactly the same type (and both are kratos-specific, so no reason to prefer one over the other).

**Comment by loumalouomega** [comment]
Thanks

**Pull request merged**